app/services/subscriptions/create_service.rb

```diff
@@ -42,21 +46,29 @@ def process_create
       return result.fail!('missing_argument', 'unable to find customer') unless current_customer
       return result.fail!('missing_argument', 'plan does not exists') unless current_plan
 
+      if currency_missmatch?(current_customer&.active_subscription&.plan, current_plan)
+        return result.fail!('currencies_does_not_match', 'currencies does not match')
+      end
+
+
       result.subscription = handle_subscription
       result
     rescue ActiveRecord::RecordInvalid => e
       result.fail_with_validations!(e.record)
     end
 
-    def current_subscription
-      @current_subscription ||= current_customer.subscriptions.active.first
+    def find_current_subscription(subscription_id)
+      return nil unless current_customer
+      return nil unless subscription_id
+
+      current_customer.subscriptions.active.find_by(id: subscription_id)
```

Comment:
You could make it shorter by doing:

```ruby
current_customer&.active_subscriptions&.find_by(id: subscription_id)
```

Or

```ruby
Subscription.active.find_by(
  id: subscription_id,
  customer_id: current_customer&.id,
)
```

Reply:
updated